bundle/src/main/java/com/adobe/acs/commons/cmf/TraversingContentMigrationStep.java

```diff
@@ -0,0 +1,60 @@
+package com.adobe.acs.commons.cmf;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+
+
+/**
+ * This class implements the Visitor pattern for the identifyResources() method.
+ *
+ */
+public abstract class TraversingContentMigrationStep implements ContentMigrationStep {
+
+	@Override
+	public List<Resource> identifyResources(Resource rootResource) {
+		
+		List<Resource> result = new LinkedList<Resource>();
+		return visit (result, rootResource);
+	}
+	
+	/**
+	 * Visits the resource r and its sub-resources. 
+	 * @param r
+	 * @return
+	 */
+	private List<Resource> visit (List<Resource> list, Resource r) {
```

Comment:
I wonder if this should have some filter options... back in 6.0 days i had to rely every heavily on traversing (search was still evolving) and I became close friends with `accepts(..)` in [1] .. i let me decrease my traversals by vast amounts since i typically didnt want to traverse below the [cq:Page] or [dam:Asset] level - once I was at those levels, the underlying structure was well-known and i didnt have traverse everything below.


[1] https://sling.apache.org/apidocs/sling7/org/apache/sling/api/resource/AbstractResourceVisitor.html